### ISSUE TITLE
Bugs in S3method dispatching and ggbiplot

### ISSUE THREAD

**Issue by psalazar80** [zdk123/compPLS#3]
The examples proposed by the author are not running and may require an update: 
1. To my knowledge, the small example 1 with ArcticLake dataset runs only after creating a new column with factor to apply the PLSDA algorithm. It currently uses the scalar continuous response which would be fine for usual pls but not for plsDA classification.
A simple workaround is to create a binary factor related to the magnitude of the current scalar response (depth) above/below median such as:
q50.arct <- quantile(ArcticLake[,4],0.50,digits =1)
for (i in 1:length(ArcticLake[,4])) {
  ArcticLake[i,4] <- ifelse(ArcticLake[i,4] <q50.arct, 1,2)
} 

After this simple change, ggbiplot worked but only using the following syntax: and not with: ggbiplot() 
compPLS::ggbiplot.plsda(res1,group.ellipse=TRUE,grouping= as.factor(ArcticLake[,4]),label.loadings=TRUE, label.offset=.2, alpha=.6)

2. The second example with 'hydrochem' dataset correctly computes the plsDA with or without bootstrapping (with plsDA_main()) but I could not apply the ggbiplot() either using  ggbiplot() or compPLS::ggbiplot.plsda() even restricting k=2 and no bootstrap like in the first dataset.

For example res.h2 below is apparently correctly computed with optK=2 but fails with ggbiplot():

res.h2 <- plsDA_main(Hchem.clr, grouping=Hydrochem$River, K=optK, nboots=10,n.core=6) # if on multicore system

compPLS::ggbiplot.plsda(res.h2, grouping=Hydrochem$River, group.ellipse=TRUE, alpha=.5, plot.loadings=FALSE, label.loadings=TRUE)

Error in sqrt(nrow(scores)) : 
  non-numeric argument to mathematical function

Thanks. The code is useful and would deserve clean working examples for broader adoption with clinical researchers who do not code on a daily basis.

**Comment by zdk123** [comment]
Thanks for the bug report!

The main problem was some change in S3method dispatching - hence code like `ggbiplot` wasn't working on a `plsda` object but `ggbiplot.plsda` was (this should work automatically).

Can you please install from the `patch` branch and run through the example code again?
```r
devtools::install_github("zdk123/compPLS", ref="patch")
```

I'll pull that in if it's working now.

The issue that's perplexing me is that we seem to have different ArcticLake datasets. I see:
```r
> ArcticLake[,4]
 [1] shallow shallow shallow shallow shallow shallow shallow shallow shallow
[10] shallow shallow shallow shallow medium  medium  medium  medium  medium
[19] medium  medium  medium  medium  medium  medium  medium  medium  medium
[28] medium  deep    deep    deep    deep    deep    deep    deep    deep
[37] deep    deep    deep
Levels: deep medium shallow
```
which is obviously categorical data.

**Comment by psalazar80** [comment]
Hi Zak, I am happy to confirm that your code is working well now!

Regarding ArcticLake[,4], I take it from the package ‘compositions’ and it has the depth (scalar) in the 4th column.

Thanks so much for your help.

Pascal

**Comment by zdk123** [comment]
I see. I actually have a copy of the ArcticLake data in this package with the categorical construct in the fourth column. Your way is fine, but importing the compositions package will hide my copy.

**Comment by zdk123** [comment]
Fixed in master by PR#4
